### DISCUSSION TITLE
tests/dummy/translations for dummy app on addon v1

### DISCUSSION THREAD

**Discussion by rgaiacs** [ember-intl/ember-intl#1904] [Q&A] [answered]
Hi,

I have some translations that I want to ship with my addon and some other translations that I only use in the dummy demo app of my addon that I don't want to ship with my addon.

What are the steps to have `translations` and `tests/dummy/translations` working together for the dummy app on addon v1?

Do I need to use a separate standalone app like described in https://ember-learn.github.io/ember-cli-addon-docs/docs/standalone-apps?

Thanks!

**Comment by ijlee2** [comment, marked as answer]
@rgaiacs Good question, I had some idea but didn't check until today.

The thing to keep in mind is, in v1 addons, the folder structure in `tests/dummy` closely matches that of an Ember app. (`dummy` is the test app that we can run locally to check the addon, without having to publish the addon.)

Since, for apps, we save translation files in the `translations` folder, this implies that translation files for `dummy` go to `tests/dummy/translations`. What's tricky is knowing how to update `inputPath` in `tests/dummy/config/ember-intl.js` (remaining keys omitted for brevity):

```diff
module.exports = function (/* environment */) {
  return {
-     inputPath: 'translations',
+     inputPath: 'tests/dummy/translations',
  };
};
```

With these 2 steps (storing translations in the right place and updating `inputPath`), you should be able to see translations from the addon as well as from `dummy`.

> [!NOTE]
>
> Since the `dummy` app is an Ember app, [translations for `dummy` can overwrite those from the addon](https://ember-intl.github.io/ember-intl/versions/v7.0.2/docs/quickstart-addons#2-add-a-translation), if that's something that you need to check.

Alternative solutions are:

1. [Pass a translations object to `setupIntl()`](https://ember-intl.github.io/ember-intl/versions/v7.0.2/docs/test-helpers/setup-intl#setupintl-hooks-locale-translations-) as needed.
2. Create a monorepo so that you can have 1 (or more) Ember app that checks your addon. The app would have its own `translations` directory.
